### ISSUE TITLE
Upload fields in groups get (sometimes) lost going back/forth - multipage form

### ISSUE THREAD

**Issue by mandrasch** [verbb/formie#1590]
### Describe the bug

Hey,

I'm chasing a very weird bug with multiple upload fields in multi page forms. 🧐 

Most of the time going back and forth works fine. But in some cases files in groups will get lost while going back - and they will also disappear on page reloads.

Reproduction video:

https://github.com/verbb/formie/assets/777278/487b9101-6a7a-4a27-a864-929d9eb37ed1

My assumption was that hitting "next" (submit) will save the files in the (incomplete) submission? Maybe this is not working properly in some case? I could not really pinpoint what is going on, maybe related to upload fields in groups? But not sure. Or the reload action to update the status of upload is not triggered?

Happy to help investigate this further!

Hope I did not miss anything.

Thanks very much!

### Steps to reproduce

1. Installed fresh craft via my template https://github.com/mandrasch/ddev-craftcms-vite
2. Install vite plugin 
3. Installed formie plugin
4. Create new form, JSON export: https://gist.github.com/mandrasch/f299580850d09cd3f6492a56b3a23621
4. Create form template to show it:

```twig
<!DOCTYPE html>
<html lang="en-US">
<head>
	<meta http-equiv="X-UA-Compatible" content="IE=edge" />
	<meta charset="utf-8" />
	<title>Welcome to Craft CMS</title>
	<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, viewport-fit=cover" />
	<meta name="referrer" content="origin-when-cross-origin" />

	{{ craft.vite.script("src/js/app.js") }}
</head>
<body class="ltr">

	<div style="display:flex; justify-content: center;margin-top:50px;">
		<div style="width:750px;padding:20px;text-align:left; border:1px solid black;">

			{{ craft.formie.renderFormCss('multiStepFormWithUploads') }}

			{{ craft.formie.renderForm('multiStepFormWithUploads') }}

			{{ craft.formie.renderFormJs('multiStepFormWithUploads') }}

		</div>
	</div>

</body>
</html>
```

### Form settings

- Multi-page form: Yes
- Submission Method: Ajax
- Client-side Validation: Yes
- Custom Form Templates: No

### Craft CMS version

4.5.5

### Plugin version

2.0.38

### Multi-site?

No

**Comment by engram-design** [comment]
Looks like this is an issue when the File Upload field is the only field in a Group/Repeater field. The inconsistent nature of it would be that it works for _new_ submissions where there is no "row" for the Group field, so one is created, and the value is considered "dirty" and the NestedFieldRow element is saved, which cascades to the File Upload field uploading the asset.

However, for existing NestedFieldRow's (when you go back and edit the field after having submitted the page), it's not detecting that the value has changed, because File Upload's don't actually send their content in the traditional sense. They're a separate bit of content through uploaded data. So Formie is looking at the existing NestedFieldRow for the Group field, seeing that no data has changed and it's empty, so it doesn't trigger a save. This is to save with expensive element-saving unless the content of a Group/Repeater field has changed.

Having another field in the Group field's nested fields would have triggered a save - like a SIngle-Line Text field, as it would detect the NestedFieldRow has content.

Long explanation I know! Fixed for the next release. To get this early, run `composer require verbb/formie:"dev-craft-4 as 2.0.38"`.

**Comment by mandrasch** [comment]
Hi @engram-design,

thanks very much for the quick fix and the insightful explanation, much appreciated! :-)

Just used  `ddev composer require verbb/formie:"dev-craft-4 as 2.0.38"` and all uploads fields worked fine. 👍 🎉

**Comment by engram-design** [comment]
Fixed in [2.0.39](https://github.com/verbb/formie/releases/tag/2.0.39)
